pkg/api/v1alpha1/cluster_types.go

```diff
@@ -323,9 +325,48 @@ func (n *ClusterNetwork) Equal(o *ClusterNetwork) bool {
 	if n == nil || o == nil {
 		return false
 	}
+
+	if n.CNI != "" {
+		if o.CNI != n.CNI {
+			return false
+		}
+	}
+
+	if n.CNIConfig != nil {
+		if o.CNI != "" {
+			switch o.CNI {
+			case Cilium, CiliumEnterprise:
+				if n.CNIConfig.Cilium == nil {
+					return false
+				}
+			case Kindnetd:
+				if n.CNIConfig.Kindnetd == nil {
+					return false
+				}
+			default:
+				return false
+			}
+		}
+		if o.CNIConfig != nil {
+			if !n.CNIConfig.Equal(o.CNIConfig) {
+				return false
+			}
+		}
+	}
+
 	return SliceEqual(n.Pods.CidrBlocks, o.Pods.CidrBlocks) &&
 		SliceEqual(n.Services.CidrBlocks, o.Services.CidrBlocks) &&
-		n.CNI == o.CNI && n.DNS.ResolvConf.Equal(o.DNS.ResolvConf)
+		n.DNS.ResolvConf.Equal(o.DNS.ResolvConf)
+}
+
+func (n *CNIConfig) Equal(o *CNIConfig) bool {
+	if (n.Cilium != nil && o.Cilium == nil) || (n.Cilium == nil && o.Cilium != nil) {
```

Comment:
This is not checking the value of `PolicyEnforcementMode`

Reply:
I left that on purpose, I think switching between different policy enforcement modes should be allowed since it's all going through the helm chart, what do you think? also @vignesh-goutham would like your opinion on this

Reply:
Oh yeah I agree with that
However, this is an `Equal` method, so it should test equality I think

Also, I think there is a nuance. We can allow to update this from the cli but not from the controller (since we don't update/upgrade the cni from the controller)

Reply:
changed this so that clusterNetwork.Equal checks the entire object including configuration options for the cni plugins like cilium.
added separate Equal checks for individual clusterNetwork fields so that CLI can call those, and allow changing policyEnforcementMode.
webhook will continue to call clusterNetwork.Equal so no changes will be allowed through the webhook